pytorch_lightning/loggers/mlflow.py

```diff
@@ -220,6 +233,28 @@ def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) ->
 
             self.experiment.log_metric(self.run_id, k, v, timestamp_ms, step)
 
+    @rank_zero_only
+    def log_figure(self, name: str, figure: plt.figure, step: Optional[int] = None, close: bool = True) -> None:
+
+        if step is not None:
+            figure_fname = f"figure_{name}_step_{step}{self._figure_file_extension}"
+        else:
+            figure_fname = f"figure_{name}{self._figure_file_extension}"
+
+        # create tmp directory and semantically named filebecause
+        # apparently one should not write to artifact location directly
+        # ToDo: Once its stable, use ml_flow.log_figure
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            figure_path = Path(tmp_dir) / figure_fname
+            figure.savefig(figure_path)
```

Comment:
Did you check the figure is properly rendered ? From past experience, `figure.savefig` never worked for me.

Reply:
Just changed the test for it to an actual functional one instead of mocking, and for me it works fine. The .png / .pdf is created as expected.
Locally it works as well. What happened in your case? The file was created but not filled?